ansible/README.md

```diff
@@ -64,47 +67,73 @@ This needs to be done only once per environment. It will install necessary prere
+
+**Caveat:** Mac users who have a docker-machine setup will have to re-run this playbook every time the boot2docker machine is rebooted. This is because installed prereqs are not persisted on TinyCore Linux.
 
 
 ### Deploying Using CouchDB
-- Uncomment the couchdb secion in `environments/<environment>/group_vars/all`
-- Comment out the cloudant section
+- Make sure the couchdb section is **uncommented** in `environments/<environment>/group_vars/all`
+- Make sure the cloudant section is **commented out**
 - Then execute
 
 ```
-ansible-playbook -i environments/<environment> couchdb.yml
-ansible-playbook -i environments/<environment> couchdb.yml -e mode=initdb
 cd <openwhisk_home>
 gradle distDocker
 cd ansible
+ansible-playbook -i environments/<environment> couchdb.yml
+ansible-playbook -i environments/<environment> couchdb.yml -e mode=initdb
 ansible-playbook -i environments/<environment> openwhisk.yml
 ```
 
 You need to run "initdb" on couch **every time** you deploy couchdb to initialize the db.
 
 ### Deploying Using Cloudant
-- Enter your cloudant credentials in `environments/<environment>/group_vars/all`
-- Comment out the couchdb section
+- Make sure the cloudant section is **uncommented** in in `environments/<environment>/group_vars/all`
+- Make sure the couchdb section is **commented out**
+- Enter your cloudant credentials in the cloudant section
 - Then execute
 
 ```
-ansible-playbook -i environments/<environment> cloudant.yml -e mode=initdb
 cd <openwhisk_home>
 gradle distDocker
 cd ansible
+ansible-playbook -i environments/<environment> cloudant.yml -e mode=initdb
 ansible-playbook -i environments/<environment> openwhisk.yml
 ```
 You need to run "initdb" on cloudant **only once** per cloudant database to initialize the db.
 
-**Hint:** In either case the `initdb` mode will ask for your confirmation to drop the database. You can skip this
-confirmation step by adding `-e prompt_user=false` to the command:
+**Hint:** In either case the `initdb` mode will ask for your confirmation to drop the database. To move on you need to enter `CTRL-C + C`. To abort you need to enter `CTRL-C + A`. You can skip this confirmation step by adding `-e prompt_user=false` to the command:
 
 ```
 ansible-playbook -i environments/<environment> couchdb.yml -e mode=initdb -e prompt_user=false
 ```
 
+### Hot-swapping a Single Component
+The playbook structure allows you to clean, deploy or re-deploy a single component as well as the entire Openwhisk stack. Let's assume you have deployed the entire stack using the "openwhisk.yml" playbook. You then make a change to a single component, for example the invoker. You will probably want a new tag on the invoker image so you first build it using:
+
+```
+cd <openwhisk_home>
+gradle :core:dispatcher:distDocker -PdockerImageTag=myNewInvoker
```

Comment:
yes, it will then replace the "latest" image instead. also mentioned that in the readme.